modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/messages/StatisticsGatheringResponse.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2020 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ignite.internal.processors.query.stat.messages;
+
+import org.apache.ignite.internal.GridDirectMap;
+import org.apache.ignite.plugin.extensions.communication.Message;
+import org.apache.ignite.plugin.extensions.communication.MessageCollectionItemType;
+import org.apache.ignite.plugin.extensions.communication.MessageReader;
+import org.apache.ignite.plugin.extensions.communication.MessageWriter;
+
+import java.io.Externalizable;
+import java.nio.ByteBuffer;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Message to send statistics.
+ */
+public class StatisticsGatheringResponse implements Message {
+    /** */
+    private static final long serialVersionUID = 0L;
+
+    /** */
+    public static final short TYPE_CODE = 186;
+
+    /** Collection id. */
+    private UUID gatId;
+
+    /** Request id. */
+    private UUID reqId;
+
+    /** Map of collected local object statistics with array of included partitions. */
+    @GridDirectMap(keyType = StatisticsObjectData.class, valueType = int[].class)
+    private Map<StatisticsObjectData, int[]> data;
```

Comment:
StatisticsObjectData has default equals and hashcode methods. IS it OK?

Reply:
Map removed with a sinle group per gathering task change